lib/pbench/test/unit/server/test_state_tracker.py

```diff
@@ -28,13 +29,14 @@ def test_construct(self):
         """ Test dataset contructor
         """
         ds = Dataset(user="drb", controller="frodo", name="fio")
+        ds.add()
         assert ds.user == "drb"
         assert ds.controller == "frodo"
         assert ds.name == "fio"
         assert ds.state == States.UPLOADING
         assert ds.md5 is None
-        assert ds.id is None
-        assert ds.created == ds.transition
+        assert ds.id is not None
+        assert ds.created <= ds.transition
         ds.add()
```

Comment:
Are you intentionally adding the same dataset twice?

Reply:
Um, no. The change from `__init__()` to the validation decorator and event listener meant that some of the tests I wanted can no longer be performed on a "raw" un-added object; and in this case I didn't notice that there was a later `add` (to test that an `id` was generated), so I failed to remove it.

Apparently SQLAlchemy is smart enough to elide the duplicate add on the same object; I'm not sure I like that, but I'm not going to sweat over it now unless you decide you're going to stuff me into a sauna about this...

Reply:
Nope, no stuffing.  Just apprising you of what I noticed.

However, I don't think the `.add()` does all that much without a `.commit()`...right?  So, presumably, you can do as many `.add()`'s as you want, building and tweaking, and then finally commit the result when it's ready.